[Bug] Build error: `sched_cls_id_` maybe uninitialized

### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.


### Ray Component

Ray Core

### What happened + What you expected to happen

1. Bug:
```
In file included from bazel-out/k8-opt/bin/_virtual_includes/ray_common/ray/common/task/task_util.h:19,
                 from bazel-out/k8-opt/bin/_virtual_includes/core_worker_lib/ray/core_worker/actor_handle.h:20,
                 from bazel-out/k8-opt/bin/_virtual_includes/core_worker_lib/ray/core_worker/core_worker.h:22,
                 from src/ray/core_worker/core_worker.cc:15:
In member function ‘ray::TaskSpecification& ray::TaskSpecification::operator=(ray::TaskSpecification&&)’,
    inlined from ‘ray::Status ray::core::CoreWorker::ExecuteTask(const ray::TaskSpecification&, const std::shared_ptr<std::unordered_map<std::__cxx11::basic_string<char>, std::vector<std::pair<long int, double> > > >&, std::vector<std::shared_ptr<ray::RayObject> >*, ray::core::ReferenceCounter::ReferenceTableProto*)’ at src/ray/core_worker/core_worker.cc:2327:39:
bazel-out/k8-opt/bin/_virtual_includes/ray_common/ray/common/task/task_spec.h:58:7: error: ‘<anonymous>.ray::TaskSpecification::sched_cls_id_’ may be used uninitialized [-Werror=maybe-uninitialized]
   58 | class TaskSpecification : public MessageWrapper<rpc::TaskSpec> {
      |       ^~~~~~~~~~~~~~~~~
src/ray/core_worker/core_worker.cc: In member function ‘ray::Status ray::core::CoreWorker::ExecuteTask(const ray::TaskSpecification&, const std::shared_ptr<std::unordered_map<std::__cxx11::basic_string<char>, std::vector<std::pair<long int, double> > > >&, std::vector<std::shared_ptr<ray::RayObject> >*, ray::core::ReferenceCounter::ReferenceTableProto*)’:
src/ray/core_worker/core_worker.cc:2327:39: note: ‘<anonymous>’ declared here
 2327 |     current_task_ = TaskSpecification();
      |                                       ^
```
2. Expected behavior: build successfully
3. sysinfo
OS: ArchLinux
GCC: 11.1.0
Bazel: 4.2.0
Java: 11.0.12+7

### Reproduction script

N/A build error

### Anything else

N/A

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

--- a/src/ray/common/task/task_spec.h
+++ b/src/ray/common/task/task_spec.h
@@ -254,7 +254,7 @@ class TaskSpecification : public MessageWrapper<rpc::TaskSpec> {
   /// Field storing required placement resources. Initialized in constructor.
   std::shared_ptr<ResourceSet> required_placement_resources_;
   /// Cached scheduling class of this task.
-  SchedulingClass sched_cls_id_;
+  SchedulingClass sched_cls_id_ = 0;
 
   /// Below static fields could be mutated in `ComputeResources` concurrently due to
   /// multi-threading, we need a mutex to protect it.

